From 6e78cf225602e33f82546ec35d2dabf7e0b3353b Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Tue, 19 Jul 2022 14:49:44 +0800
Subject: [PATCH 15/62] Revert "drm/bridge: analogix_dp: Support PSR-exit to
 disable transition"

This reverts commit dbe04e874d4fbd56be64fdfcb29410241b6ad08a.
---
 .../drm/bridge/analogix/analogix_dp_core.c    | 42 ++-----------------
 1 file changed, 4 insertions(+), 38 deletions(-)

diff --git a/drivers/gpu/drm/bridge/analogix/analogix_dp_core.c b/drivers/gpu/drm/bridge/analogix/analogix_dp_core.c
index a7bcb429c02b..9755672caf1a 100644
--- a/drivers/gpu/drm/bridge/analogix/analogix_dp_core.c
+++ b/drivers/gpu/drm/bridge/analogix/analogix_dp_core.c
@@ -1268,25 +1268,6 @@ static int analogix_dp_bridge_attach(struct drm_bridge *bridge,
 	return 0;
 }
 
-static
-struct drm_crtc *analogix_dp_get_old_crtc(struct analogix_dp_device *dp,
-					  struct drm_atomic_state *state)
-{
-	struct drm_encoder *encoder = dp->encoder;
-	struct drm_connector *connector;
-	struct drm_connector_state *conn_state;
-
-	connector = drm_atomic_get_old_connector_for_encoder(state, encoder);
-	if (!connector)
-		return NULL;
-
-	conn_state = drm_atomic_get_old_connector_state(state, connector);
-	if (!conn_state)
-		return NULL;
-
-	return conn_state->crtc;
-}
-
 static
 struct drm_crtc *analogix_dp_get_new_crtc(struct analogix_dp_device *dp,
 					  struct drm_atomic_state *state)
@@ -1467,16 +1448,14 @@ analogix_dp_bridge_atomic_disable(struct drm_bridge *bridge,
 {
 	struct drm_atomic_state *old_state = old_bridge_state->base.state;
 	struct analogix_dp_device *dp = bridge->driver_private;
-	struct drm_crtc *old_crtc, *new_crtc;
-	struct drm_crtc_state *old_crtc_state = NULL;
+	struct drm_crtc *crtc;
 	struct drm_crtc_state *new_crtc_state = NULL;
-	int ret;
 
-	new_crtc = analogix_dp_get_new_crtc(dp, old_state);
-	if (!new_crtc)
+	crtc = analogix_dp_get_new_crtc(dp, old_state);
+	if (!crtc)
 		goto out;
 
-	new_crtc_state = drm_atomic_get_new_crtc_state(old_state, new_crtc);
+	new_crtc_state = drm_atomic_get_new_crtc_state(old_state, crtc);
 	if (!new_crtc_state)
 		goto out;
 
@@ -1485,19 +1464,6 @@ analogix_dp_bridge_atomic_disable(struct drm_bridge *bridge,
 		return;
 
 out:
-	old_crtc = analogix_dp_get_old_crtc(dp, old_state);
-	if (old_crtc) {
-		old_crtc_state = drm_atomic_get_old_crtc_state(old_state,
-							       old_crtc);
-
-		/* When moving from PSR to fully disabled, exit PSR first. */
-		if (old_crtc_state && old_crtc_state->self_refresh_active) {
-			ret = analogix_dp_disable_psr(dp);
-			if (ret)
-				DRM_ERROR("Failed to disable psr (%d)\n", ret);
-		}
-	}
-
 	analogix_dp_bridge_disable(bridge);
 }
 
-- 
2.27.0

